src/Models/HydrostaticFreeSurfaceModels/split_explicit_free_surface.jl

```diff
@@ -245,9 +239,35 @@ struct ForwardBackwardScheme end
 end
 
 @inline cosine_averaging_kernel(τ::FT) where FT = τ >= 0.5 && τ <= 1.5 ? FT(1 + cos(2π * (τ - 1))) : zero(FT)
-
 @inline constant_averaging_kernel(τ) = 1
 
+struct AdaptiveSubsteps{B, F}
+    Δtᴮ :: B
+    averaging_kernel :: F
+end
+
+struct FixedSubsteps{B, F}
+    Δτ :: B
+    averaging_weights :: F
+end
```

Comment:
More verbose names might help here... I guess `Δtᴮ` is `Δt_barotropic` ? Not exactly sure what `Δτ` is but maybe its a fractional step size?

Another improvement might be `AdaptiveSplitExplicitSubsteps` and `FixedSplitExplicitSubsteps`.

However, in the "fixed" case, it's just the _number_ that's fixed right? Not the step size?vThat isn't clear from the name, but could maybe just be fixed by a docstring.